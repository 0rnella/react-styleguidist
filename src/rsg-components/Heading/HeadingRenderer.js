import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Slot from 'rsg-components/Slot';
import Styled from 'rsg-components/Styled';

export const styles = ({ color, space, fontSize, fontFamily }) => ({
	root: {
		display: 'flex',
		marginBottom: space[1],
		alignItems: 'center',
	},
	heading: {
		color: color.base,
		fontSize: fontSize.h2,
		fontFamily: fontFamily.base,
		fontWeight: 'normal',
	},
	isPrimary: {
		fontSize: fontSize.h1,
	},
	isDeprecated: {
		textDecoration: 'line-through',
		color: color.light,
	},
	toolbar: {
		marginLeft: 'auto',
	},
});

export function HeadingRenderer({
	classes,
	children,
	id,
	slotName,
	slotProps,
	primary,
	deprecated,
}) {
	const Tag = primary ? 'h1' : 'h2';
	const headingClasses = cx(classes.heading, {
		[classes.isPrimary]: primary,
		[classes.isDeprecated]: deprecated,
	});
	return (
		<Tag id={id} className={classes.root}>
			<div className={headingClasses}>{children}</div>
			<Slot className={classes.toolbar} name={slotName} props={slotProps} />
		</Tag>
	);
}

HeadingRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	children: PropTypes.node,
	id: PropTypes.string.isRequired,
	slotName: PropTypes.string.isRequired,
	slotProps: PropTypes.object.isRequired,
	primary: PropTypes.bool,
	deprecated: PropTypes.bool,
};

export default Styled(styles)(HeadingRenderer);
