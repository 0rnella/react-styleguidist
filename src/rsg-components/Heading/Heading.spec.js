import React from 'react';
import { HeadingRenderer, styles } from './HeadingRenderer';

const props = {
	classes: classes(styles),
	id: 'heading',
	slotName: 'pizza',
	slotProps: {},
};

it('renderer should render H2 tag', () => {
	const actual = shallow(<HeadingRenderer {...props}>Heading</HeadingRenderer>);

	expect(actual).toMatchSnapshot();
});

it('renderer should render H1 tag', () => {
	const actual = shallow(<HeadingRenderer {...props} primary>Heading</HeadingRenderer>);

	expect(actual).toMatchSnapshot();
});

it('renderer should render heading with deprecated styles', () => {
	const actual = shallow(<HeadingRenderer {...props} deprecated>Heading</HeadingRenderer>);

	expect(actual).toMatchSnapshot();
});

it('renderer should pass props to slots', () => {
	const actual = shallow(
		<HeadingRenderer {...props} slotName="slot" slotProps={{ foo: 1, bar: 'baz' }}>
			Heading
		</HeadingRenderer>
	);

	expect(actual).toMatchSnapshot();
});
