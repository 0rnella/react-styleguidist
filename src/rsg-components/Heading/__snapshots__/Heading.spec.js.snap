// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should pass props to slots 1`] = `
<h2
  className="root"
  id="heading"
>
  <div
    className="heading"
  >
    Heading
  </div>
  <Slot
    className="toolbar"
    name="slot"
    props={
      Object {
        "bar": "baz",
        "foo": 1,
      }
    }
  />
</h2>
`;

exports[`renderer should render H1 tag 1`] = `
<h1
  className="root"
  id="heading"
>
  <div
    className="heading isPrimary"
  >
    Heading
  </div>
  <Slot
    className="toolbar"
    name="pizza"
    props={Object {}}
  />
</h1>
`;

exports[`renderer should render H2 tag 1`] = `
<h2
  className="root"
  id="heading"
>
  <div
    className="heading"
  >
    Heading
  </div>
  <Slot
    className="toolbar"
    name="pizza"
    props={Object {}}
  />
</h2>
`;

exports[`renderer should render heading with deprecated styles 1`] = `
<h2
  className="root"
  id="heading"
>
  <div
    className="heading isDeprecated"
  >
    Heading
  </div>
  <Slot
    className="toolbar"
    name="pizza"
    props={Object {}}
  />
</h2>
`;
